packages/app-desktop/gui/ErrorBoundary.tsx

```diff
@@ -61,9 +61,13 @@ export default class ErrorBoundary extends React.Component<Props, State> {
 
 	public componentDidMount() {
 		const onAppClose = () => {
-			ipcRenderer.send('asynchronous-message', 'appCloseReply', {
-				canClose: true,
-			});
+			// Replies to the appClose event are generally handled elsewhere. However,
+			// if we're in an error state, these handlers aren't guranteed to work.
+			if (this.state.error) {
+				ipcRenderer.send('asynchronous-message', 'appCloseReply', {
```

Comment:
An `appCloseReply` is also sent [in MainScreen.tsx](https://github.com/laurent22/joplin/blob/ddc83ec15370884fb8d06bd2cef44544cfac33e9/packages/app-desktop/gui/MainScreen/MainScreen.tsx#L280). The other reply is only sent after the promise returned by `Setting.saveAll` resolves.

Thus, the reply sent by `ErrorBoundary.tsx` would arrive first, causing the app to close, before the `wasClosedSuccessfully` setting could be saved.